pkg/util/ipset/ipset.go

```diff
@@ -0,0 +1,72 @@
+package ipset
+
+import (
+	"github.com/pkg/errors"
+	"os/exec"
+)
+
+type Name string
+
+type Type string
+
+const (
+	ListSet = Type("list:set")
+	HashIP  = Type("hash:ip")
+)
+
+type Interface interface {
+	Create(ipsetName Name, ipsetType Type) error
+	AddEntry(ipsetName Name, entry string) error
+	DelEntry(ipsetName Name, entry string) error
+	Flush(ipsetName Name) error
+	Destroy(ipsetName Name) error
+
+	FlushAll() error
+	DestroyAll() error
+}
+
+type ipset struct {
+}
+
+func New() Interface {
```

Comment:
This package is modelled on `k8s.io/kubernetes/pkg/util/iptables` so I've followed the convention there.